python/cudf_polars/tests/test_scan.py

```diff
@@ -487,3 +491,72 @@ def test_scan_from_file_uri(tmp_path: Path) -> None:
     df.write_parquet(path)
     q = pl.scan_parquet(f"file://{path}")
     assert_ir_translation_raises(q, NotImplementedError)
+
+
+@pytest.mark.parametrize("chunked", [False, True])
+def test_scan_parquet_remote(
+    tmp_path: Path, df: pl.DataFrame, httpserver: HTTPServer, *, chunked: bool
+) -> None:
+    path = tmp_path / "foo.parquet"
+    df.write_parquet(path)
+    bytes_ = path.read_bytes()
+    size = len(bytes_)
+
+    def head_handler(_: Request) -> Response:
+        return Response(
+            status=200,
+            headers={
+                "Content-Type": "parquet",
+                "Accept-Ranges": "bytes",
+                "Content-Length": size,
+            },
+        )
+
+    def get_handler(req: Request) -> Response:
+        # parse bytes=200-500 for example (the actual data)
+        start, end = map(int, req.headers["Range"][6:].split("-"))
+        mv = memoryview(bytes_)[start : end + 1]
+        return Response(
+            mv.tobytes(),
+            status=200,
+            headers={
+                "Content-Type": "parquet",
+                "Accept-Ranges": "bytes",
+                "Content-Length": len(mv),
+                "Content-Range": f"bytes {start}-{end - 1}/{size}",
+            },
+        )
+
+    httpserver.expect_request(path, method="HEAD").respond_with_handler(head_handler)
+    httpserver.expect_request(path, method="GET").respond_with_handler(get_handler)
+
+    q = pl.scan_parquet(path)
```

Comment:
This is probably me misunderstanding things, but I would expect a test like `q = pl.scan_parquet("http://<server-address>/<file-relative-to-server-root>")`. Right now, it seems like we're still using the filesystem rather than remote IO via the HTTP server. I think it'd be something like

```
q = pl.scan_parquet(httpserver.url_for(path))
```

To make the two paths (filesystem & server) clearer, we might want to change this to something like

```
server_path = "foo.parquet"
httpserver.expect_request(server_path, method="HEAD").respond_with_handler(head_handler)
httpserver.expect_request(server_path, method="GET").respond_with_handler(get_handler)

q = pl.scan_parquet(httpserver.url_for(serer_path))
```